answers.txt.txt

```diff
@@ -0,0 +1,26 @@
+1. How does your method for extracting features work?
+
+First, we check if the bed file contains the chrom from the  query file. 
+If  it does, we procees to choose just the lines from bed file that are between the start and end position,
+according to line in query file. Then, we print those lines to our outfile.
+
+
+
+2. What is the complexity of the algorithm, as a function of the size of the two input files?
+When you answer this, you need to know that you can get the list of chromosomse from a query.Table in constant time, 
+but it does, of course, take longer to run through all the lines in it.
+
+First, we iterate through the lines of bed file, to extract them to the table O(n). 
+Then, we iterate through all the lines in the query table O(m), there are three actions in the loop :
+split the line O(1), get_chrom O(1) and then we run through the every line in bed to print the lines that agree with out condition O(n). 
+This means, our overall complexity is O(n*m)
+
+
+3. Did you, at any point, exploit that our features are on single nucleotides and not larger regions?
+
+The algorithm does not make assumptions about single nucleotides. 
```

Comment:
Do you want to bet? I feel pretty confident that this code will break if I give it general feature intervals:

```python
 if bed_line.chrom_start >= int(lst[1]) and bed_line.chrom_end <= int(lst[2]): 
                 print_line(bed_line, args.outfile)
```

Let's say the query interval was `[5, 10)` (from five up to, but not including, 10).

Let's also say that I had features for the intervals `[1, 20)`, `[3, 6)`, `[6, 8)`, `[8, 25)`. All four intervals overlap the query, so they should all be included. But are they?

```
[1, 20) -- (bedline.chrom_start >= int(lst[1])) == (1 >= 5) == False -- not included
[3, 6)  -- (bedline.chrom_start >= int(lst[1])) == (3 >= 5) == False -- not included
[6, 8)  -- (bedline.chrom_start >= int(lst[1])) == (6 >= 5) == True
           (bedline.chrom_end <= int(lst[2])) == (8 <= 10) == True -- This one is included
[8, 25) -- (bedline.chrom_start >= int(lst[1])) == (8 >= 5) == True
           (bedline.chrom_end <= int(lst[2])) == (25 <= 10) == False -- not included
```
Your code finds 25% of the features it is supposed to here.

Generally, you will find all features that are completely contained in the query interval, and if features have length one, then that is all the features that overlap. With general ranges, though, you can overlap in ways that you do not catch.

You don't have to fix this, but please give it a try to work out what would be necessary to find all overlapping intervals in the general case.